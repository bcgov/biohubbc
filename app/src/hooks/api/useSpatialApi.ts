import { AxiosInstance } from 'axios';
import { Feature } from 'geojson';

export type RegionDetails = { regionName: string; sourceLayer: string };
export type GetRegionsResponse = { regions: RegionDetails[] };

/**
 * Returns a set of supported api methods for working with spatial information.
 *
 * @param {AxiosInstance} axios
 * @return {*} object whose properties are supported api methods.
 */
const useSpatialApi = (axios: AxiosInstance) => {
  /**
   * Fetch a list of BCGW region names that intersect the provided features.
   *
   * @param {Feature[]} features
   * @return {*}  {Promise<{ regions: GetRegionsResponse[] }>}
   */
  const getRegions = async (features: Feature[]): Promise<GetRegionsResponse> => {
    const { data } = await axios.post('/api/spatial/regions', { features: features });

    return data;
  };

  return {
    getRegions
  };
};

export default useSpatialApi;
