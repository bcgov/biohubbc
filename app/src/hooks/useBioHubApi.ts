import { useKeycloak } from '@react-keycloak/web';
import axios from 'axios';
import { IProject, IProjectPostObject } from 'interfaces/project-interfaces';
import { IActivity, ICreateActivity, ICreateProjectResponse, ITemplate } from 'interfaces/useBioHubApi-interfaces';
import moment from 'moment';
import { useMemo } from 'react';

const API_HOST = process.env.REACT_APP_API_HOST;
const API_PORT = process.env.REACT_APP_API_PORT;

const API_URL = (API_PORT && `${API_HOST}:${API_PORT}`) || API_HOST || 'api-dev-biohubbc.apps.silver.devops.gov.bc.ca';

/**
 * Returns an instance of axios with baseURL and authorization headers set.
 *
 * @return {*}
 */
const useApi = () => {
  const { keycloak } = useKeycloak();
  const instance = useMemo(() => {
    return axios.create({
      headers: {
        Authorization: `Bearer ${keycloak?.token}`
      },
      baseURL: API_URL
    });
  }, [keycloak]);

  return instance;
};

/**
 * Returns a set of supported api methods.
 *
 * @return {object} object whose properties are supported api methods.
 */
export const useBiohubApi = () => {
  const api = useApi();

  /**
   * Get all projects.
   *
   * @return {*}  {Promise<IProject[]>}
   */
  const getProjects = async (): Promise<IProject[]> => {
    // const { data } = await api.get(`/api/projects`);

    // return data;

    // TODO: stub for development
    return [
      {
        id: 1,
        name: 'Project Name 1',
        objectives: 'Project Objectives 1',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 1',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 1',
        caveats: 'Caveats 1',
        comments: 'Comments 1'
      },
      {
        id: 2,
        name: 'Project Name 2',
        objectives: 'Project Objectives 2',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 2',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 2',
        caveats: 'Caveats 2',
        comments: 'Comments 2'
      },
      {
        id: 3,
        name: 'Project Name 3',
        objectives: 'Project Objectives 3',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 3',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 3',
        caveats: 'Caveats 3',
        comments: 'Comments 3'
      },
      {
        id: 4,
        name: 'Project Name 4',
        objectives: 'Project Objectives 4',
        scientific_collection_permit_number: '123456',
        management_recovery_action: 'A',
        location_description: 'Location Description 4',
        start_date: moment().toISOString(),
        end_date: moment().toISOString(),
        results: 'Results 4',
        caveats: 'Caveats 4',
        comments: 'Comments 4'
      }
    ];
  };

  /**
   * Get a project based on its ID.
   *
   * @param {projectId} projectId
   * @return {*}  {Promise<IProject>}
   */
  const getProject = async (projectId: number): Promise<IProject> => {
    const { data } = await api.get(`/api/project/${projectId}`);

    return data;
  };

  /**
   * Create a new project.
   *
   * @param {IProjectPostObject} project
   * @return {*}  {Promise<ICreateProjectResponse>}
   */
  const createProject = async (project: IProjectPostObject): Promise<ICreateProjectResponse> => {
    const { data } = await api.post('/api/project', project);

    return data;
  };

  /**
   * Get a template based on its ID.
   *
   * @param {templateId} templateId
   * @return {*}  {Promise<ITemplate>}
   */
  const getTemplate = async (templateId: number): Promise<ITemplate> => {
    const { data } = await api.get(`/api/template/${templateId}`);

    return data;
  };

  /**
   * Create a new activity record.
   *
   * @param {ICreateActivity} activity
   * @return {*}  {Promise<IActivity>}
   */
  const createActivity = async (activity: ICreateActivity): Promise<IActivity> => {
    const { data } = await api.post('/api/activity', activity);

    return data;
  };

  /**
   * Fetch all code sets.
   *
   * @return {*}  {Promise<any>}
   */
  const getAllCodes = async (): Promise<any> => {
    const { data } = await api.get('/api/codes/');

    return data;
  };

  /**
   * Fetch the api json-schema spec.
   *
   * @return {*}  {Promise<any>}
   */
  const getApiSpec = async (): Promise<any> => {
    const { data } = await api.get('/api/api-docs/');

    return data;
  };

  return {
    getProjects,
    getProject,
    getTemplate,
    createProject,
    createActivity,
    getAllCodes,
    getApiSpec
  };
};
