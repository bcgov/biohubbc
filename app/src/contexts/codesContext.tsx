import { useBiohubApi } from 'hooks/useBioHubApi';
import useDataLoader, { DataLoader } from 'hooks/useDataLoader';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import { createContext, PropsWithChildren, useMemo } from 'react';

/**
 * Context object that stores information about codes
 *
 * @export
 * @interface ICodesContext
 */
export interface ICodesContext {
  /**
   * The Data Loader used to load codes
   *
   * @type {DataLoader<[], IGetAllCodeSetsResponse, unknown>}
   * @memberof ICodesContext
   */
  codesDataLoader: DataLoader<[], IGetAllCodeSetsResponse, unknown>;
}

export const CodesContext = createContext<ICodesContext>({
  codesDataLoader: {} as DataLoader<never, IGetAllCodeSetsResponse, unknown>
});

export const CodesContextProvider = (props: PropsWithChildren<Record<never, any>>) => {
  const biohubApi = useBiohubApi();
  const codesDataLoader = useDataLoader(biohubApi.codes.getAllCodeSets);

  const codesContext: ICodesContext = useMemo(() => ({ codesDataLoader }), [codesDataLoader]);

  return <CodesContext.Provider value={codesContext}>{props.children}</CodesContext.Provider>;
};
