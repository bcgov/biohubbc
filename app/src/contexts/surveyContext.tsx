import { useBiohubApi } from 'hooks/useBioHubApi';
import useDataLoader, { DataLoader } from 'hooks/useDataLoader';
import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
import React, { createContext, PropsWithChildren, useEffect, useMemo } from 'react';
import { useParams } from 'react-router';

/**
 * Context object that stores information about a survey
 *
 * @export
 * @interface ISurveyContext
 */
export interface ISurveyContext {
  /**
   * The Data Loader used to load survey data
   *
   * @type {DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>}
   * @memberof ISurveyContext
   */
  surveyDataLoader: DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>;

  /**
   * The project ID belonging to the current project
   *
   * @type {number}
   * @memberof ISurveyContext
   */
  projectId: number;

  /**
   * The ID belonging to the current survey
   *
   * @type {number}
   * @memberof ISurveyContext
   */
  surveyId: number;
}

export const SurveyContext = createContext<ISurveyContext>({
  surveyDataLoader: {} as DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>,
  projectId: -1,
  surveyId: -1
});

export const SurveyContextProvider = (props: PropsWithChildren<Record<never, any>>) => {
  const biohubApi = useBiohubApi();
  const surveyDataLoader = useDataLoader(biohubApi.survey.getSurveyForView);
  const urlParams = useParams();

  if (!urlParams['id']) {
    throw new Error(
      "The project ID found in SurveyContextProvider was invalid. Does your current React route provide an 'id' parameter?"
    );
  }

  if (!urlParams['survey_id']) {
    throw new Error(
      "The survey ID found in SurveyContextProvider was invalid. Does your current React route provide a 'survey_id' parameter?"
    );
  }

  const projectId = Number(urlParams['id']);
  const surveyId = Number(urlParams['survey_id']);

  /**
   * Refreshes the current survey object whenever the current survey ID changes
   */
  useEffect(() => {
    if (projectId && surveyId) {
      surveyDataLoader.refresh(projectId, surveyId);
    }

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [projectId, surveyId]);

  const surveyContext: ISurveyContext = useMemo(
    () => ({
      surveyDataLoader,
      projectId,
      surveyId
    }),
    [surveyDataLoader, projectId, surveyId]
  );

  return <SurveyContext.Provider value={surveyContext}>{props.children}</SurveyContext.Provider>;
};
