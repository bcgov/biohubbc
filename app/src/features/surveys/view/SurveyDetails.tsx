import Box from '@material-ui/core/Box';
import Paper from '@material-ui/core/Paper';
import { Theme } from '@material-ui/core/styles/createMuiTheme';
import Typography from '@material-ui/core/Typography';
import makeStyles from '@material-ui/core/styles/makeStyles';
import SurveyGeneralInformation from 'features/surveys/view/components/SurveyGeneralInformation';
import SurveyProprietaryData from 'features/surveys/view/components/SurveyProprietaryData';
import React, { useContext } from 'react';
import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';
import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import SurveyStudyArea from './components/SurveyStudyArea';
import Button from '@material-ui/core/Button';
import Icon from '@mdi/react';
import { mdiDelete } from '@mdi/js';
import { DialogContext } from 'contexts/dialogContext';

export interface ISurveyDetailsProps {
  surveyForViewData: IGetSurveyForViewResponse;
  codes: IGetAllCodeSetsResponse;
  projectForViewData: IGetProjectForViewResponse;
  refresh: () => void;
}

const useStyles = makeStyles((theme: Theme) => ({
  surveyDetailsSection: {
    marginTop: theme.spacing(4),
    marginBottom: theme.spacing(5),
    '&:last-child': {
      marginBottom: 0
    },
    '&:first-child': {
      marginTop: 0
    }
  }
}));

/**
 * Details content for a survey.
 *
 * @return {*}
 */
const SurveyDetails: React.FC<ISurveyDetailsProps> = (props) => {
  const { surveyForViewData, codes, refresh, projectForViewData } = props;

  const classes = useStyles();
  const dialogContext = useContext(DialogContext);

  const defaultYesNoDialogProps = {
    dialogTitle: 'Delete Survey',
    dialogText: 'Are you sure you want to delete this survey, its attachments and associated observations?',
    open: false,
    onClose: () => dialogContext.setYesNoDialog({ open: false }),
    onNo: () => dialogContext.setYesNoDialog({ open: false }),
    onYes: () => dialogContext.setYesNoDialog({ open: false })
  };

  const showDeleteSurveyDialog = () => {
    dialogContext.setYesNoDialog({
      ...defaultYesNoDialogProps,
      open: true,
      onYes: () => {
        deleteSurvey();
        dialogContext.setYesNoDialog({ open: false });
      }
    });
  };

  const deleteSurvey = () => {
    console.log('delete survey');
  };

  return (
    <>
      <Box mb={5} display="flex" justifyContent="space-between">
        <Typography variant="h2">Survey Details</Typography>
        <Button
          variant="outlined"
          color="primary"
          startIcon={<Icon path={mdiDelete} size={1} />}
          onClick={showDeleteSurveyDialog}>
          Delete Survey
        </Button>
      </Box>

      <Box component={Paper} p={4}>
        <Box component="section" className={classes.surveyDetailsSection}>
          <SurveyGeneralInformation
            projectForViewData={projectForViewData}
            surveyForViewData={surveyForViewData}
            codes={codes}
            refresh={refresh}
          />
        </Box>
      </Box>

      <Box component={Paper} p={4} mt={4}>
        <Box component="section" className={classes.surveyDetailsSection}>
          <SurveyStudyArea
            surveyForViewData={surveyForViewData}
            projectForViewData={projectForViewData}
            refresh={refresh}
          />
        </Box>
      </Box>

      <Box component={Paper} p={4} mt={4}>
        <Box component="section" className={classes.surveyDetailsSection}>
          <SurveyProprietaryData
            projectForViewData={projectForViewData}
            surveyForViewData={surveyForViewData}
            codes={codes}
            refresh={refresh}
          />
        </Box>
      </Box>
    </>
  );
};

export default SurveyDetails;
