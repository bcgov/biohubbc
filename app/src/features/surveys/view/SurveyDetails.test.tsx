import { CodesContext, ICodesContext } from 'contexts/codesContext';
import { SurveyContext } from 'contexts/surveyContext';
import { createMemoryHistory } from 'history';
import { GetRegionsResponse } from 'hooks/api/useSpatialApi';
import { DataLoader } from 'hooks/useDataLoader';
import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
import { Router } from 'react-router';
import { codes } from 'test-helpers/code-helpers';
import { getSurveyForViewResponse } from 'test-helpers/survey-helpers';
import { render, waitFor } from 'test-helpers/test-utils';
import { useBiohubApi } from '../../../hooks/useBioHubApi';
import SurveyDetails from './SurveyDetails';

const history = createMemoryHistory({ initialEntries: ['/admin/projects/1/surveys/2'] });

jest.mock('../../../hooks/useBioHubApi');
const mockBiohubApi = useBiohubApi as jest.Mock;

const mockUseApi = {
  spatial: {
    getRegions: jest.fn<Promise<GetRegionsResponse>, []>()
  }
};

describe('SurveyDetails', () => {
  beforeEach(() => {
    mockBiohubApi.mockImplementation(() => mockUseApi);
    mockUseApi.spatial.getRegions.mockClear();

    mockUseApi.spatial.getRegions.mockResolvedValue({
      regions: []
    });
  });

  const mockCodesContext: ICodesContext = {
    codesDataLoader: {
      data: codes
    } as DataLoader<any, any, any>
  };

  const mockSurveyDataLoader = {
    data: {
      ...getSurveyForViewResponse,
      surveyData: {
        ...getSurveyForViewResponse.surveyData,
        purpose_and_methodology: {
          ...getSurveyForViewResponse.surveyData.purpose_and_methodology,
          additional_details: '' // no additional details
        }
      }
    }
  } as DataLoader<any, IGetSurveyForViewResponse, any>;
  const mockArtifactDataLoader = { data: null } as DataLoader<any, any, any>;
  const mockSampleSiteDataLoader = { data: null } as DataLoader<any, any, any>;
  const mockCritterDataLoader = { data: [] } as DataLoader<any, any, any>;
  const mockDeploymentDataLoader = { data: [] } as DataLoader<any, any, any>;

  it('renders correctly', async () => {
    const { getByText } = render(
      <Router history={history}>
        <SurveyContext.Provider
          value={{
            projectId: 1,
            surveyId: 1,
            critterDeployments: [],
            surveyDataLoader: mockSurveyDataLoader,
            artifactDataLoader: mockArtifactDataLoader,
            sampleSiteDataLoader: mockSampleSiteDataLoader,
            critterDataLoader: mockCritterDataLoader,
            deploymentDataLoader: mockDeploymentDataLoader
          }}>
          <CodesContext.Provider value={mockCodesContext}>
            <SurveyDetails />
          </CodesContext.Provider>
        </SurveyContext.Provider>
      </Router>
    );

    await waitFor(() => {
      expect(getByText('Survey Details')).toBeInTheDocument();
    });
  });
});
