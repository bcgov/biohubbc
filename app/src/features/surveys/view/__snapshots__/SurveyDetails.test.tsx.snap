// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2"
    >
      Survey Details
    </h2>
  </div>
  <div
    class="MuiPaper-root MuiBox-root MuiBox-root-3 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <section
      class="MuiBox-root MuiBox-root-4 makeStyles-surveyDetailsSection-1"
    >
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            General Information
          </h3>
          <button
            aria-label="Edit Survey General Information"
            class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
            tabindex="0"
            title="Edit Survey General Information"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.3125rem; height: 1.3125rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Edit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <dl>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Survey Name
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                survey name
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Survey Timeline
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                October 10, 1998 - February 26, 2021
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Survey Lead
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                first last
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Focal Species
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                focal species 1
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Anciliary Species
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                ancillary species 1
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Survey Methodology
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                method
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Permit
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                123 - Scientific
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Funding Sources
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Funding Agency Blah | $ 100 | April 09, 2000 - May 10, 2000
              </dd>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiBox-root MuiBox-root-7"
              >
                <div
                  class="MuiBox-root MuiBox-root-8"
                >
                  <dt
                    class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                  >
                    Purpose
                  </dt>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                >
                  survey purpose
                </p>
              </div>
            </div>
          </div>
        </dl>
      </div>
    </section>
  </div>
  <div
    class="MuiPaper-root MuiBox-root MuiBox-root-9 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <section
      class="MuiBox-root MuiBox-root-10 makeStyles-surveyDetailsSection-1"
    >
      <div
        class="MuiBox-root MuiBox-root-11"
      >
        <div
          class="MuiBox-root MuiBox-root-12"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Study Area
          </h3>
          <button
            aria-label="Edit Study Area"
            class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
            tabindex="0"
            title="Edit Study Area"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.3125rem; height: 1.3125rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Edit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <dl>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Survey Area Name
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                study area
              </dd>
            </div>
          </div>
        </dl>
        <div
          class="MuiBox-root MuiBox-root-13"
        >
          <div
            class="leaflet-container leaflet-grab leaflet-touch-drag"
            id="survey_study_area_map"
            style="height: 100%; position: relative;"
            tabindex="0"
          >
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://maps.gov.bc.ca/arcgis/rest/services/province/roads_wm/MapServer/tile/0/0/0"
                      style="width: 256px; height: 256px; left: -38px; top: -79px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              >
                <svg
                  height="0"
                  pointer-events="none"
                  style="left: 0px; top: 0px;"
                  viewBox="0 0 0 0"
                  width="0"
                >
                  <g>
                    <path
                      class="leaflet-interactive"
                      d="M-1 2L0 -2z"
                      fill="#3388ff"
                      fill-opacity="0.2"
                      fill-rule="evenodd"
                      stroke="#3388ff"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-opacity="1"
                      stroke-width="3"
                    />
                  </g>
                </svg>
              </div>
              <div
                class="leaflet-pane leaflet-marker-pane"
              />
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              >
                <div
                  class="leaflet-control-zoom leaflet-bar leaflet-control"
                >
                  <a
                    aria-label="Zoom in"
                    class="leaflet-control-zoom-in"
                    href="#"
                    role="button"
                    title="Zoom in"
                  >
                    +
                  </a>
                  <a
                    aria-label="Zoom out"
                    class="leaflet-control-zoom-out leaflet-disabled"
                    href="#"
                    role="button"
                    title="Zoom out"
                  >
                    −
                  </a>
                </div>
                <div
                  class="leaflet-control-fullscreen leaflet-bar leaflet-control"
                >
                  <a
                    class="leaflet-control-fullscreen-button leaflet-bar-part"
                    href="#"
                    title="View Fullscreen"
                  />
                </div>
              </div>
              <div
                class="leaflet-top leaflet-right"
              >
                <div
                  class="leaflet-draw leaflet-control"
                />
                <div
                  class="leaflet-draw leaflet-control"
                />
              </div>
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  aria-haspopup="true"
                  class="leaflet-control-layers leaflet-control"
                >
                  <a
                    class="leaflet-control-layers-toggle"
                    href="#"
                    title="Layers"
                  />
                  <section
                    class="leaflet-control-layers-list"
                  >
                    <div
                      class="leaflet-control-layers-base"
                    >
                      <label>
                        <div>
                          <input
                            checked="checked"
                            class="leaflet-control-layers-selector"
                            name="leaflet-base-layers_36"
                            type="radio"
                          />
                          <span>
                             BC Government
                          </span>
                        </div>
                      </label>
                      <label>
                        <div>
                          <input
                            class="leaflet-control-layers-selector"
                            name="leaflet-base-layers_36"
                            type="radio"
                          />
                          <span>
                             Esri Imagery
                          </span>
                        </div>
                      </label>
                    </div>
                    <div
                      class="leaflet-control-layers-separator"
                      style="display: none;"
                    />
                    <div
                      class="leaflet-control-layers-overlays"
                    />
                  </section>
                </div>
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          />
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          />
        </div>
      </div>
    </section>
  </div>
  <div
    class="MuiPaper-root MuiBox-root MuiBox-root-14 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <section
      class="MuiBox-root MuiBox-root-15 makeStyles-surveyDetailsSection-1"
    >
      <div
        class="MuiBox-root MuiBox-root-16"
      >
        <div
          class="MuiBox-root MuiBox-root-17"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Proprietary Data
          </h3>
          <button
            aria-label="Edit Proprietary Data"
            class="MuiButtonBase-root MuiButton-root MuiButton-text sectionHeaderButton MuiButton-textPrimary"
            tabindex="0"
            title="Edit Proprietary Data"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  role="presentation"
                  style="width: 1.3125rem; height: 1.3125rem;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14.06,9L15,9.94L5.92,19H5V18.08L14.06,9M17.66,3C17.41,3 17.15,3.1 16.96,3.29L15.13,5.12L18.88,8.87L20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18.17,3.09 17.92,3 17.66,3M14.06,6.19L3,17.25V21H6.75L17.81,9.94L14.06,6.19Z"
                    style="fill: currentColor;"
                  />
                </svg>
              </span>
              Edit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <dl>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Proprietary Data Category
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                proprietor type
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Proprietor Name
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                prop name
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
            >
              <dt
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
              >
                Data and Information Sharing Agreement Required
              </dt>
              <dd
                class="MuiTypography-root MuiTypography-body1"
              >
                Yes
              </dd>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiBox-root MuiBox-root-18"
              >
                <div
                  class="MuiBox-root MuiBox-root-19"
                >
                  <dt
                    class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                  >
                    Category Rationale
                  </dt>
                </div>
                <p
                  class="MuiTypography-root MuiTypography-body1"
                  style="word-break: break-all;"
                >
                  rationale
                </p>
              </div>
            </div>
          </div>
        </dl>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
